services/src/pro/layers/postgres_layer_db.rs

```diff
@@ -331,6 +334,123 @@ where
+
+        // delete the collection!
+        // on delete cascade removes all entries from `collection_children` and `collection_layers`
+
+        let remove_layer_collection_stmt = conn
+            .prepare(
+                "DELETE FROM layer_collections
+                 WHERE id = $1;",
+            )
+            .await?;
+        conn.execute(&remove_layer_collection_stmt, &[&collection])
+            .await?;
+
+        // delete all collections without parent collection
+        // TODO: use recursive delete statement
+
+        let remove_layer_collections_without_parents_stmt = conn
+            .prepare(
+                "DELETE FROM layer_collections
+                 WHERE  id <> $1 -- do not delete root collection
+                 AND    id NOT IN (
+                    SELECT child FROM collection_children
+                 );",
+            )
+            .await?;
+        while 0 < conn
+            .execute(
+                &remove_layer_collections_without_parents_stmt,
+                &[&INTERNAL_LAYER_DB_ROOT_COLLECTION_ID],
+            )
+            .await?
+        {
+            // whenever one collection is deleted, we have to check again if there are more
+            // collections without parents
+        }
+
+        // delete all layers without parent collection
+
+        let remove_layers_without_parents_stmt = conn
+            .prepare(
+                "DELETE FROM layers
+                 WHERE id NOT IN (
+                    SELECT layer FROM collection_layers
+                 );",
+            )
+            .await?;
+        conn.execute(&remove_layers_without_parents_stmt, &[])
+            .await?;
+
+        Ok(())
+    }
+
+    async fn remove_layer_from_collection(
+        &self,
+        layer: &LayerId,
+        collection: &LayerCollectionId,
+    ) -> Result<()> {
+        let collection_uuid =
+            Uuid::from_str(&collection.0).map_err(|_| crate::error::Error::IdStringMustBeUuid {
+                found: collection.0.clone(),
+            })?;
+
+        let layer_uuid =
+            Uuid::from_str(&layer.0).map_err(|_| crate::error::Error::IdStringMustBeUuid {
+                found: layer.0.clone(),
+            })?;
+
+        let conn = self.conn_pool.get().await?;
+
+        let remove_layer_collection_stmt = conn
+            .prepare(
+                "DELETE FROM collection_layers
+                 WHERE collection = $1
+                 AND layer = $2;",
+            )
+            .await?;
+        let num_results = conn
+            .execute(
+                &remove_layer_collection_stmt,
+                &[&collection_uuid, &layer_uuid],
+            )
+            .await?;
+
+        if num_results == 0 {
+            return Err(LayerDbError::NoLayerForGivenIdInCollection {
+                layer: layer.clone(),
+                collection: collection.clone(),
+            }
+            .into());
+        }
+
+        // remove layers without any collection
```

Comment:
why do you need this here? you do not delete any collections.

Reply:
[331a3ca](https://github.com/geo-engine/geoengine/pull/676/commits/331a3ca526cd68e4039441ab5faf1fc0e52b3515)

It is not. When you remove a layer from a collection, it could have been the only collection of that layer.
So, you need try clean up the layer if it is without any collection reference.